src/web/tools/KeyEventManager.ts

```diff
@@ -0,0 +1,103 @@
+import { AdaptedEvent, EventTypes } from '../interfaces';
+import EventManager from './EventManager';
+import { PointerType } from '../../PointerType';
+import { View } from 'react-native';
+
+export default class KeyEventManager extends EventManager<HTMLElement> {
+  private activationKeys = ['Enter', 'Space', 'NumpadEnter'];
+  private cancelationKeys = ['Tab'];
+
+  private keyDownCallback = (event: KeyboardEvent): void => {
+    console.log(event.code);
+    if (this.cancelationKeys.indexOf(event.code) !== -1) {
+      this.dispatchEvent(event, EventTypes.CANCEL);
+    }
+
+    if (this.activationKeys.indexOf(event.code) === -1) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.DOWN);
+  };
+
+  private keyUpCallback = (event: KeyboardEvent): void => {
+    if (this.activationKeys.indexOf(event.code) === -1) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.UP);
+  };
+
+  private dispatchEvent(event: KeyboardEvent, eventType: EventTypes) {
+    (event.target as unknown as View)?.measure?.(
+      (_x, _y, w, h, pageX, pageY) => {
+        const adaptedEvent = this.adaptEvent(
+          event,
+          eventType,
+          pageX,
+          pageY,
+          w,
+          h
+        );
```

Comment:
Wouldn't it be possible to somehow use `mapEvent` instead of adding another method?

Reply:
I agree that we should use that instead of adding a new method. Measurement on web can be synchronous so that logic can be moved there so different signature shouldn't be a problem.

Reply:
As of [1573a90](https://github.com/software-mansion/react-native-gesture-handler/pull/3035/commits/1573a9039ff361ae950150bcd70c22333aee9c33) I moved measurement logic to `mapEvent` and removed `adaptEvent` all together.